### PR TITLE
[document-picker][android][ios][web] Allow multiple mimetypes and fix bug

### LINKED ISSUE #4263: [document-picker] Passing multiple mime types on document picker
I'm trying to get spreadsheet format files only but seems it is currenty not possible. It would be nice to have this. Also is it possible to use some wildcard for showing spreadsheets format only? I tried `application/*` but its not working.

### LINKED ISSUE #8029: DocumentPicker isn't allowing all file types when browsing iCloud Drive locations
We've got an Expo app that we're currently preparing for release and we use Expo `DocumentPicker` during the onboarding process. We've discovered that when browsing iCloud Drive locations only a subset of files/types are selectable. As far as I can tell we've setup everything correctly according to the [Configuration](https://docs.expo.io/versions/latest/sdk/document-picker/#configuration).

The `.keystore` files we're using (and some other file types eg. `.sketch`) our greyed out and unselectable, but only on iCloud Drive.

- The file we're trying to select is a JSON document using `.keystore` extension.
- Everything works correctly on Android (whether using phone local locations or remote locations)
- Everything works correctly on iPhone when picking local files or from other remote locations (eg. Google Drive)
- We can pick other files from iCloud Drive (eg. images or PDF's), they come into our app and the app has some logic to report that they're invalid files.

Am I missing some required config? Is this a bug? Any help is much appreciated

### PR DESCRIPTION
**Pull request by mstach60161**
# Why

- Fixes #4263.
- Fixes #8029.

# How

- Added possibility to pass array of mimetypes.
- Changed UTType for "*/*" mimetype from public.data to public.item. [IOS]

# Test Plan

- Native component list.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).

### CODE REVIEW

File: packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m [hunk 1]
```diff
@@ -111,13 +119,19 @@ - (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
     @try {
       // TODO: drop #if macro once Xcode is updated to 12
 #if __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000
-      if (@available(iOS 14, *)) {
-        UTType* utType = EXConvertMimeTypeToUTType(mimeType);
-        documentPickerVC = [[UIDocumentPickerViewController alloc] initForOpeningContentTypes:@[utType] asCopy:YES];
+      if ((@available(iOS 14, *))) {
```
Comment (bbarthec) [inline packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m:122]:
```suggestion
      if (@available(iOS 14, *)) {
```

File: packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m [hunk 2]
```diff
@@ -111,13 +119,19 @@ - (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
...
+        NSMutableArray *uttArray = [mimeTypes mutableCopy];
```
Comment (bbarthec) [inline packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m:123]: We prefer verbosity 👍 
It's always good to omit almost all abbreviations in open source software, or even in software altogether 😉 
```suggestion
        NSMutableArray *utTypes = [mimeTypes mutableCopy];
```

File: packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m [hunk 3]
```diff
@@ -111,13 +119,19 @@ - (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
...
+        for (int i = 0; i < [mimeTypes count]; i++) {
+          uttArray[i] = (NSString *)EXConvertMimeTypeToUTType(mimeTypes[i]);
```
Comment (bbarthec) [inline packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m:125]:
```suggestion
          utTypes[i] = (NSString *)EXConvertMimeTypeToUTType(mimeTypes[i]);
```

File: packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m [hunk 4]
```diff
@@ -111,13 +119,19 @@ - (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
...
+        }
+        documentPickerVC = [[UIDocumentPickerViewController alloc] initForOpeningContentTypes:uttArray asCopy:true];
```
Comment (bbarthec) [inline packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m:127]:
```suggestion
        documentPickerVC = [[UIDocumentPickerViewController alloc] initForOpeningContentTypes:utTypes asCopy:true];
```

**Review by bbarthec — CHANGES_REQUESTED** [review]
Looks fine to me 👍 
Things to do before we merge:
- [ ] Apply CI suggestion for CHANGELOG
- [ ] Back away from using abbreviations that could be full variables names

File: packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerOptions.kt [hunk 5]
```diff
@@ -2,27 +2,30 @@ package expo.modules.documentpicker
 
 import org.unimodules.core.Promise
 
-data class DocumentPickerOptions(val type: String, val copyToCacheDirectory: Boolean) {
+data class DocumentPickerOptions(val type: String, val copyToCacheDirectory: Boolean, val types: Array<String>?) {
```
Comment (lukmccall) [inline packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerOptions.kt:5]: Do we have `types` and `type`? Can we merge it?

**Review by lukmccall — CHANGES_REQUESTED** [review]
I'm not a big fan of passing string or array to the native side. I think it will be better to wrap a single string into an array in js. In that way, we don't have to check the type of this argument on the native side.

File: packages/expo-document-picker/src/ExpoDocumentPicker.web.ts [hunk 6]
```diff
@@ -20,7 +20,7 @@ export default {
     const input = document.createElement('input');
     input.style.display = 'none';
     input.setAttribute('type', 'file');
-    input.setAttribute('accept', type);
+    input.setAttribute('accept', type as string);
```
Comment (bbarthec) [inline packages/expo-document-picker/src/ExpoDocumentPicker.web.ts:23]: Here we've got a problem 🤔 
This is the available typing: `type: string | string[]` and you narrow it down to `string` without any check. What about `string[]`? Is it impossible to pass `string[]` on the web? 🤔 If so we need to document this and properly handle `string[]` here 🤔

**Review by bbarthec — COMMENTED** [review]
I believe something is wrong in the native iOS and Android code 🤔 
I see that you've changed the ReactNative native method signature in the JS realm to `getDocumentAsync({ type: string[], ... }`, but on the native level you handle both cases `string | string[]`. I believe we could simplify to `string[]`.

File: packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m [hunk 7]
```diff
@@ -112,12 +117,18 @@ - (void)setModuleRegistry:(EXModuleRegistry *)moduleRegistry
...
+          utTypes[i] = (NSString *)EXConvertMimeTypeToUTType(mimeTypes[i]);
+        }
+        documentPickerVC = [[UIDocumentPickerViewController alloc] initForOpeningContentTypes:utTypes asCopy:true];
       } else {
 #endif
-        NSString* type = EXConvertMimeTypeToUTI(mimeType);
-        documentPickerVC = [[UIDocumentPickerViewController alloc] initWithDocumentTypes:@[type]
+        NSMutableArray *utiTypes = [mimeTypes mutableCopy];
+        for (int i = 0; i < [mimeTypes count]; i++) {
+            utiTypes[i] = (NSString *)EXConvertMimeTypeToUTI(mimeTypes[i]);
```
Comment (bbarthec) [inline packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m:129]:
```suggestion
          utiTypes[i] = (NSString *)EXConvertMimeTypeToUTI(mimeTypes[i]);
```

**Review by bbarthec — APPROVED** [review]
It looks neat now! 😊

**Comment by bbarthec** [comment]
Fix the CI, `yarn build` and we're ready to merge it 😉

**Pull request merged**